ansible/roles/prep.dirs/tasks/repo.yml

```diff
@@ -4,38 +4,40 @@
     insertafter: EOF
     line: "include('{{ mock_dir }}/glusterfs-nightly-repo.tpl')"
   with_items:
-    - "{{ mock_dir }}/centos-stream+epel-next-{{ item }}-x86_64.cfg"
-    - "{{ mock_dir }}/rhel-{{ item }}-x86_64.cfg"
+    - "{{ mock_dir }}/centos-stream+epel-next-{{ os_vers }}-x86_64.cfg"
+    - "{{ mock_dir }}/rhel-{{ os_vers }}-x86_64.cfg"
   loop_control:
     loop_var: mock_config
 
 - name: adapt mock config to use additional template for ceph
   block:
-    - name: determine ceph repo base url
-      shell: curl -s "https://shaman.ceph.com/api/search/?project=ceph&distros=centos/{{ item }}/x86_64&flavor=default&ref=main&sha1=latest" | jq -r .[0].url
-      register: get_ceph_repo_url
+    - block:
+      - name: determine ceph repo base url
+        shell: curl -s "https://shaman.ceph.com/api/search/?project=ceph&distros=centos/{{ os_vers }}/x86_64&flavor=default&ref=main&sha1=latest" | jq -r .[0].url
+        register: get_ceph_repo_url
 
-    - set_fact:
-        "ceph_el{{ item }}_repo_base_url": "{{ get_ceph_repo_url.stdout }}"
+      - set_fact:
+          "ceph_base_url": "{{ get_ceph_repo_url.stdout }}"
+      when: not ceph_base_url
```

Comment:
I would recommend `when: ceph_base_url is not defined` from your previous iteration. See related comment towards the end.